src/Compilers/CSharp/Portable/Symbols/MergedNamespaceSymbol.cs

```diff
@@ -82,14 +83,30 @@ internal sealed class MergedNamespaceSymbol : NamespaceSymbol
 
             Debug.Assert(namespacesToMerge.Length != 0);
 
-            return (namespacesToMerge.Length == 1 && nameOpt == null)
-                ? namespacesToMerge[0]
-                : new MergedNamespaceSymbol(extent, containingNamespace, namespacesToMerge, nameOpt);
+            if (namespacesToMerge.Length == 1 && nameOpt is null)
+                return namespacesToMerge[0];
+
+            if (namespacesToMerge.Any(ns => ns is MergedNamespaceSymbol))
+            {
+                // Resolving extern alias directives across multiple assemblies can produce multiple levels of nesting.
+                // Make sure to flatten them.
+                namespacesToMerge = namespacesToMerge.SelectManyAsArray(ns =>
+                {
+                    if (ns is MergedNamespaceSymbol merged)
+                        return merged._namespacesToMerge;
+                    else
+                        return SpecializedCollections.SingletonCollection(ns);
+                });
```

Comment:
📝 In theory, this seems to break the caching provided by `MergedNamespaceSymbol`. In practice, I wasn't able to find a way to hit this path that wasn't super obscure. It required two or more assembly references to have a shared non-default alias _and_ at least one of those references to be a netmodule with two or more winmd namespaces under it _and_ at least one of the references to not be a netmodule.